src/Underscore.php

```diff
@@ -43,6 +43,35 @@
  */
 class Underscore
 {
+    /** @var Registry */
+    protected static $registry;
+
+    /**
+     * Set the registry singleton.
+     *
+     * @param  Registry $registry
+     * @return void
+     */
+    public static function setRegistry(Registry $registry)
+    {
+        static::$registry = $registry;
```

Comment:
I don't love the singleton pattern but I think this is an appropriate usage of it. It does allow for dependency injectors to overload if necessary.